test/docker-e2e/e2e_test.go

```diff
@@ -40,78 +37,33 @@ type CelestiaTestSuite struct {
 	s.logger = zaptest.NewLogger(s.T())
-	s.logger.Info("Setting up Celestia test suite")
-	s.client, s.network = celestiadockertypes.DockerSetup(s.T())
-}
-
-type ConfigOption func(*celestiadockertypes.Config)
-
-func (s *CelestiaTestSuite) CreateDockerProvider(opts ...ConfigOption) celestiatypes.Provider {
-	numValidators := 1
-	numFullNodes := 0
-
-	enc := testutil.MakeTestEncodingConfig(app.ModuleEncodingRegisters...)
-
-	cfg := celestiadockertypes.Config{
-		Logger:          s.logger,
-		DockerClient:    s.client,
-		DockerNetworkID: s.network,
-		ChainConfig: &celestiadockertypes.ChainConfig{
-			ConfigFileOverrides: map[string]any{
-				"config/app.toml": validatorStateSyncAppOverrides(),
-			},
-			Type:          "cosmos",
-			Name:          "celestia",
-			Version:       getCelestiaTag(),
-			NumValidators: &numValidators,
-			NumFullNodes:  &numFullNodes,
-			ChainID:       "celestia",
-			Images: []celestiadockertypes.DockerImage{
-				{
-					Repository: getCelestiaImage(),
-					Version:    getCelestiaTag(),
-					UIDGID:     "10001:10001",
-				},
-			},
-			Bin:                 "celestia-appd",
-			Bech32Prefix:        "celestia",
-			Denom:               "utia",
-			CoinType:            "118",
-			GasPrices:           "0.025utia",
-			GasAdjustment:       1.3,
-			EncodingConfig:      &enc,
-			AdditionalStartArgs: []string{"--force-no-bbr", "--grpc.enable", "--grpc.address", "0.0.0.0:9090", "--rpc.grpc_laddr=tcp://0.0.0.0:9099"},
-		},
-	}
-
-	for _, opt := range opts {
-		opt(&cfg)
-	}
-
-	return celestiadockertypes.NewProvider(cfg, s.T())
+	s.logger.Info("Setting up Celestia test suite: " + s.T().Name())
+	s.client, s.network = tastoradockertypes.DockerSetup(s.T())
 }
 
 // CreateTxSim deploys and starts a txsim container to simulate transactions against the given celestia chain in the test environment.
-func (s *CelestiaTestSuite) CreateTxSim(ctx context.Context, chain celestiatypes.Chain) {
+func (s *CelestiaTestSuite) CreateTxSim(ctx context.Context, chain tastoratypes.Chain) {
 	t := s.T()
 	networkName, err := getNetworkNameFromID(ctx, s.client, s.network)
 	s.Require().NoError(err)
 
 	// Deploy txsim image
 	t.Log("Deploying txsim image")
-	txsimImage := celestiadockertypes.NewImage(s.logger, s.client, networkName, t.Name(), txsimImage, txSimTag)
+	txsimImage := tastoradockertypes.NewImage(s.logger, s.client, networkName, t.Name(), txsimImage, txSimTag)
 
-	opts := celestiadockertypes.ContainerOptions{
+	opts := tastoradockertypes.ContainerOptions{
 		User: "0:0",
 		// Mount the Celestia home directory into the txsim container
 		// this ensures txsim has access to a keyring and is able to broadcast transactions.
 		Binds: []string{chain.GetVolumeName() + ":/celestia-home"},
 	}
 
+	hostGRPCPort := strings.Split(chain.GetGRPCAddress(), ":")[1]
+
 	args := []string{
 		"/bin/txsim",
 		"--key-path", "/celestia-home",
-		"--grpc-endpoint", chain.GetGRPCAddress(),
+		"--grpc-endpoint", "host.docker.internal:" + hostGRPCPort,
```

Comment:
not 100% sure exactly what was going on here, but the host name wasn't resolving correctly without using the special DNS entry.